### DISCUSSION TITLE
Using AssetsHandler

### DISCUSSION THREAD

**Discussion by tataDan** [wailsapp/wails#1549] [Q&A]
Does anyone know of an existing example of using AssetsHandler?  The objective is to be able to create a image (e.g. a png file) on the fly and then update it and have what is displayed on the screen reflect the changes.

**Comment by stffabi** [comment]
[VarlyApp](https://github.com/varlyapp/varlyapp/blob/5b12062c8e8081ddecaaac00d353bb738f61b74f/main.go#L41) uses AssetsHandler for loading images. You might take that as a starting point.

**Reply by leaanthony** [reply]
I've added some documentation around this [here](http://wails.io/docs/next/guides/dynamic-assets). (It's still being deployed)

**Comment by tataDan** [comment]
I believe that my AssetsHandler is working properly thanks to the above suggestions.
The image name is `lineChart.png` and it is in the `frontend/src/assets/images` directory.
If in my .svelte file I use this: 
```
<img
            id="line-chart"
            src="@/assets/images/lineChart.png"
            alt="No line chart found"
        />
```
then an error such as `🚨  open /@/assets/lineChart.png: no such file or directory` will appear in the terminal output and the image does not appear on the screen.

However, if I use the full path to the png file instead of a relative path, no error appears in the terminal output and the png file appears on the screen.

Any suggestions on how I could use the relative path instead of the full path would be appreciated.

**Comment by tataDan** [comment]
I found a solution regarding the path issue that works for my application but might not be suitable for other applications.

In the app.go function where the png file is created I added the following code:
```
workDir, err := os.Getwd()
pathToLineChartFile := filepath.Join(workDir, f.Name())
return pathToLineChartFile
```
In my .svelte file I used this:
```
<img
            id="line-chart"
            src={pathToLineChartFile} alt="No line chart found" />
            alt="No line chart found"
        />
```
